src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs

```diff
@@ -245,11 +245,23 @@ private string CreateDirectoryIfDoesntExist(string targetPath)
             return fullPath;
         }
 
-        private async Task<int> GetPipelineIdAsync(AgentTaskPluginExecutionContext context, string pipelineDefinition, string pipelineVersionToDownload, string project, string[] tagFilters, string branchName = null)
+        private async Task<int> GetPipelineIdAsync(AgentTaskPluginExecutionContext context, string pipelineDefinition, string pipelineVersionToDownload, string project, string[] tagFilters, string branchName = null, CancellationToken cancellationToken = default(CancellationToken))
         {
-            var definitions = new List<int>() { Int32.Parse(pipelineDefinition) };
+            if(String.IsNullOrWhiteSpace(pipelineDefinition)) 
+            {
+                throw new InvalidOperationException("Pipeline definition cannot be null or empty");
+            }
+
             VssConnection connection = context.VssConnection;
             BuildHttpClient buildHttpClient = connection.GetClient<BuildHttpClient>();
+
+            var isDefinitionNum = Int32.TryParse(pipelineDefinition, out int definition);
+            if(!isDefinitionNum) 
+            {
+                definition = buildHttpClient.GetDefinitionsAsync(new System.Guid(project), pipelineDefinition, cancellationToken: cancellationToken).SyncResult().FirstOrDefault().Id;
```

Comment:
You can use Guid.Parse() method instead- https://docs.microsoft.com/en-us/dotnet/api/system.guid.parse?view=netframework-4.8 .

Reply:
Thanks. Looks like these two method are pretty equivalent. https://stackoverflow.com/questions/6915966/guid-parse-or-new-guid-whats-the-difference